### DISCUSSION TITLE
Max3485 pins

### DISCUSSION THREAD

**Discussion by MartinJSa** [cmcgerty/Sofar2mqtt#24] [Q&A]
Hi, I was having some stability issues with the Max485 board so had got hold of a Max3485. I'm a bit confused on the connections as they're labelled differently to the blue 485 board I've got, could someone offer some guidance on what I connect to where please?

![16532491751108876915190575378645](https://user-images.githubusercontent.com/100225653/169713416-3fc744c5-cb37-4955-9135-4f6aa5dc4f9e.jpg)

**Comment by MartinJSa** [comment]
Although, the chip on this board is a max13487EESA which according to the datasheet needs 5v input? Not the 3.3v board I thought I was getting

**Comment by cmcgerty** [comment]
Hi,

You just need the four pins on the left to go to your ESP8266. Ignore the wire from D5 to DR/RE on the 485 as the 3485 is auto-switching and doesn’t need flow control pins.

Finally, take wires A and B from the terminals on the right to you inverter.

Colin


On 22 May 2022, at 20:54, Martin Saunders ***@***.***> wrote:

﻿

Hi, I was having some stability issues with the Max485 board so had got hold of a Max3485. I'm a bit confused on the connections as they're labelled differently to the blue 485 board I've got, could someone offer some guidance on what I connect to where please?

[16532491751108876915190575378645]<https://user-images.githubusercontent.com/100225653/169713416-3fc744c5-cb37-4955-9135-4f6aa5dc4f9e.jpg>

**Comment by MartinJSa** [comment]
Thanks Colin. I've got TXD wired to D6 and RXD wired to D7 which works well for me. I'm still getting the occasional "Timeout waiting for RS485 response. Response too short" but it's pretty infrequent.

**Comment by zulufoxtrot** [comment]
Bumping this thread, because I am using the same board and I can't get it to work.

The console outputs this in a loop:
```
Timeout waiting for RS485 response.
Response too short
```
(I did at some point see some data in the MQTT topic, but it was garbled/partial/made no sense to me)

On the Max3485 board, only the TXD LED illuminates.

This is my wiring:

![EE6F0D06-E1A9-40D1-AA30-8EB02F47C263](https://github.com/user-attachments/assets/2834659b-4091-4f94-8b18-9681f12e3274)
<img src="https://github.com/user-attachments/assets/0a4a81c8-3621-4585-81f2-44c278ced3a5" width="300" />

This is the supposed pinout of my Wemos D1 mini:

<img width="1094" height="723" alt="image" src="https://github.com/user-attachments/assets/eef60375-1807-49be-9bd4-0411eefe9d58" />

The Max3485 is wired like this:
- TXD to D6 (GPIO12)
- RXD to D7 (GPIO13)


The code:
```c
#define RXPin        D6  // Serial Receive pin
#define TXPin        D7  // Serial Transmit pin
```
If I use D6 as RXpin and D7 as TXpin:

- both leds illuminate
- the inverter shows a "RS485" logo on the screen

If I use D7 as RXpin and D6 as TXpin:

- only the TX LED illuminates
- no logo on the inverter screen

In both cases, I still get the "Response too short" error.

Inverter settings (IDK if they apply to the 485s port)
- Address: 01
- Baudrate: 9600
- Databits: 8
- Parity: EVEN
- Stopbits: 1

Any help appreciated, thanks :)

**Reply by cliffdude** [reply]
Try powering the max board VCC with the 5v pin of the wemos. These boards are also somtime faulty so bare that in mind especialy if if came from aliexpress, also get more than 1 unit!

**Reply by zulufoxtrot** [reply]
I tried your advice, but I get the same result sadly...
I'll order new boards.
Thanks for the help!

**Reply by cliffdude** [reply]
That fact that you get the RS486 icon on the display is usually a sign that the board was able to talk to the inverter. Check that you have the correct Wemos board setup in the arduino IDE as that will convert D6 to GPIO?? You could use the pin number instead of D6 and D7 etc.. just to be sure.

**Reply by zulufoxtrot** [reply]
I use arduino-cli.
It has the following profiles:
```
LOLIN(WEMOS) D1 ESP-WROOM-02    esp8266:esp8266:d1_wroom_02
LOLIN(WEMOS) D1 R2 & mini       esp8266:esp8266:d1_mini
LOLIN(WEMOS) D1 mini (clone)    esp8266:esp8266:d1_mini_clone
LOLIN(WEMOS) D1 mini Lite       esp8266:esp8266:d1_mini_lite
LOLIN(WEMOS) D1 mini Pro        esp8266:esp8266:d1_mini_pro
LOLIN(WeMos) D1 R1              esp8266:esp8266:d1
```
I tried with `d1_mini` and `d1_mini_clone`, but the result is always the same :/

**Reply by zulufoxtrot** [reply]
I will try your suggestion of using the pin number 👍 
edit: didn't work, I give up
